app/models/mixins/retirement_mixin.rb

```diff
@@ -15,7 +15,10 @@ def retire(ids, options = {})
         object = find_by(:id => id)
         object.retire(options) if object.respond_to?(:retire)
       end
-      MiqQueue.put(:class_name => 'RetirementManager', :method_name => 'check')
+      q_options = {:class_name => 'RetirementManager', :method_name => 'check'}
+      user = User.current_user
```

Comment:
@lfu How do we ensure that the User.current_user is set? Should we be raising an error if its not set

Reply:
@mkanoor We only take and store the user info into the queue only when the user is set. The retirement requester would default to ```system``` if no user is specified.

Reply:
The `system` account would be used when retirement is initiated from the scheduler.